scripts/character_spawner.gd

```diff
@@ -0,0 +1,46 @@
+extends Node3D
+
+var shotgun_monk := preload("res://Scenes/Characters/shotgun_monk.tscn")
+var player_character := preload("res://Scenes/player.tscn")
+
+func _on_main_sequence_spawn_characters_signal():
+	var character = null
+	var character_x_offset : float
+	var character_y_offset : float
+	var character_z_offset : float
+	
+	character = player_character
+	character_x_offset = 8.0
+	character_y_offset = 2.0
+	character_z_offset = 8.0
+	spawn_character_instance(character, character_x_offset, character_y_offset, character_z_offset)
+	
+	character = shotgun_monk
+	character_x_offset = 8.5
+	character_y_offset = 0.0
+	character_z_offset = -8.5
+	spawn_character_instance(character, character_x_offset, character_y_offset, character_z_offset)
```

Comment:
There is a fair bit of redundant coded here, and the relevant data could just be passed directly into the two calls:
```
func _on_main_sequence_spawn_characters_signal():
	spawn_character_instance(player_character, 8.0, 2.0, 8.0)
	spawn_character_instance(shotgun_monk, 8.5, 0.0, -8.5)
```

But either this way or your original way the 'magic numbers' are not ideal, and could be made into descriptive const declarations at the top of the file.

Reply:
TBH this script was some throw-away crap to get player and NPC instantiated in the world space. It will be superceded in the next major update, so I'll leave it in this less than desirable form for now.